### DISCUSSION TITLE
Tasmota variable, state of valve --> Binary sensor in HA

### DISCUSSION THREAD

**Discussion by Saxin** [arendst/Tasmota#20157] [Q&A] [answered]
Hi,

I have device with some rules running,  3 temp sensors + 2 relay for valve control, valve is heat exchanger bypass in domestic water heating system.
Valve can be open or closed correspond to mem1 variable in rules. (0 - Closed, 1 - Open)
The logic of operation is working I just want the last bit of keeping track of valve state in HA.

I've followed this guide by @blakadder [https://blakadder.com/pir-in-tasmota-integration/](url)
Added auto discovery on System#Boot:
`ON System#boot DO publish2 homeassistant/binary_sensor/%macaddr%_hxvalve/config {"name":"hxValve","state_topic":"stat/%topic%/valve","availability_topic":"tele/%topic%/LWT","payload_available":"Online","payload_not_available":"Offline","unique_id":"%macaddr%_hxvalve","device":{"connections":[["mac","%macaddr%"]]}} ENDON`

This event is publishing the value:
ON event#mqttSend DO IF (mem1==0) publish stat/%topic%/valve OFF ELSE publish stat/%topic%/valve ON ENDIF ENDON

I can see the topic correctly published on server.
![MQTT_valve](https://github.com/arendst/Tasmota/assets/3322218/be32f88c-4c0c-4ab7-95de-317fa1c2328f)

In HA the entity appears under the corresponding device but it's status **UNKNOWN**
![HA_unknown](https://github.com/arendst/Tasmota/assets/3322218/e91146b6-8279-40e8-abd2-31eb9b85410b)

Thanks in advance.

**Comment by barbudor** [comment]
You are using a rule with IF ... ELSE ... ENDIF
As per the documentation this feature is NOT included by default in any standard binary
Have you compiled your own binary following the instructions ?

https://tasmota.github.io/docs/Rules/#conditional-rules

**Comment by Saxin** [comment]
Hi thanks it's ESP32 device, so I think it was included in default build.

**Reply by barbudor** [reply]
Ah, yes in that case, it is. 

It seems the publication is correct.
No idea what's wrong on the HA side

**Comment by blakadder** [comment, marked as answer]
HA is case sensitive on values and your MQTT explorer somehow shows "Off" while the rule is supposed to send "OFF"

**Reply by Saxin** [reply]
Thanks a lot guys, I've changed the topic to capital:
`ON event#mqttSend DO IF (mem1==0) publish stat/%topic%/VALVE OFF ELSE publish stat/%topic%/VALVE ON ENDIF ENDON`
Note the capital **VALVE**
Restarted HA and the Tasmota device and all is good it works as expected.
Thanks again much appreciated. 😃

**Reply by coltc50** [reply]
Hello, I try something similar with Home Assistant, but the event mqttSend doesn't trigger. Is this event in latest versions working? I need a trigger when switch state change in switchmode 15.

**Reply by Saxin** [reply]
> Hello, I try something similar with Home Assistant, but the event mqttSend doesn't trigger. Is this event in latest versions working? I need a trigger when switch state change in switchmode 15.

This event is defined in a rule I have, it's not built-in event in Tasmota. Take a look here [https://tasmota.github.io/docs/Rules/#examples-of-available-triggers](url). 

To give you some context here are first two lines of rule1:
```
rule1
  ON event#mqttSend DO IF (mem1==0) publish stat/%topic%/VALVE OFF ELSE publish stat/%topic%/VALVE ON ENDIF ENDON
  ON tele-DS18B20-3#temperature DO backlog event t3=%value%; event mqttSend ENDON
```

As you can see on second line when sensor3 tele-message is triggered two events are fired: t3 (which uses sensor value) and mqttSend  for publishing current state of "valve".

If you want further help, I recommend going to Tasmota discord [https://discord.gg/4MMe9qdZ](url) there is a channel dedicated to rules, people there are very nice and helpful.

Best of Luck, Eddi

**Reply by coltc50** [reply]
Thank you for quick reply... But I found another way to solve it. I take the tele topic of tasmota as state topic in hass.

Switch8 is my switchmode 15 virtual switch in tasmota.

`rule2 ON System#boot DO publish homeassistant/binary_sensor/%macaddr%_windowsz/config {"name":"window","state_topic":"tele/%topic%/SENSOR","value_template":"{{ value_json.Switch8 }}","availability_topic":"tele/%topic%/LWT","payload_available":"Online","payload_not_available":"Offline","unique_id":"%macaddr%_windowsz","device":{"connections":[["mac","%macaddr%"]]}} ENDON`
